File: src/main/java/cn/techfan/quickdoc/common/entities/FsEntity.java
package cn.techfan.quickdoc.common.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@AllArgsConstructor
@ToString
@Data
public class FsEntity {
    @Id
    private String id;
    private String filename;
    private Long contentLength;
    private String contentType;
    private Date uploadDate;
    private Long categoryId;
    private Long directoryId;
    private ObjectId storedId = null;
    private FsOwner[] owners = null;
    private String category = null;
    private String directory = null;
}

File: src/main/kotlin/cn/techfan/quickdoc/common/entities/FsEntity.kt
package cn.techfan.quickdoc.common.entities

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document
data class FsEntity(@Id val id: String,
                    var filename: String,
                    var contentLength: Long,
                    var contentType: String,
                    var uploadDate: Date,
                    var categoryId: Long,
                    var directoryId: Long,
                    var storedId: ObjectId?=null,
                    var owners: Array<FsOwner>? = null,
                    var category:String?=null,
                    var directory:String?=null)
